aws/resource_aws_kinesis_stream_consumer.go

```diff
@@ -0,0 +1,229 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsKinesisStreamConsumer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsKinesisStreamConsumerCreate,
+		Read:   resourceAwsKinesisStreamConsumerRead,
+		Delete: resourceAwsKinesisStreamConsumerDelete,
+
+		Timeouts: &schema.ResourceTimeout{
```

Comment:
generally, we prefer to handle timeout settings within the status and waiter functions on create/delete in place of operator-defined timeouts